### PR TITLE
[tests-only] Log requests and responses for the failing tests

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
This PR adds the logging feature to the API tests. All the requests and responses from scenario start to end will be logged in a file and accumulated to a failed log file if the scenario fails. And the logs can be viewed in the CI pipeline if the pipeline has a test failure.

Screenshot from 2023-10-10 09-49-40 | Screenshot from 2023-10-10 09-51-22
--|--
view logs in CI | logs example

We do have options to debug api calls using `DEBUG_ACCEPTANCE_API_CALLS` but it logs on the fly which will clutter the test logs. And we want to log only the calls that were performed during the scenario that fails and is not expected to fail.

## Motivation and Context

Get an overall view of requests and responses whenever there is API test failure. Helps in debugging flaky tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/TestHelpers/HttpLogger.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
+<?php declare(strict_types=1);
+/**
+ * ownCloud
+ *
+ * @author Sajan Gurung <sajan@jankaritech.com>
+ * @copyright Copyright (c) 2017 Sajan Gurung sajan@jankaritech.com
```
Comment (SwikritiT) [inline tests/TestHelpers/HttpLogger.php:6]: Someone's living in the past 😉 
```suggestion
 * @copyright Copyright (c) 2023 Sajan Gurung sajan@jankaritech.com
```

Comment (SwikritiT) [inline tests/TestHelpers/HttpLogger.php:6]:
```suggestion
 * @copyright Copyright (c) 2023 Sajan Gurung sajan@jankaritech.com
```

File: .drone.star [hunk 2]
```diff
@@ -2817,3 +2819,17 @@ def tikaService():
             "wait-for -it tika:9998 -t 300",
         ],
     }]
+
+def logRequests():
+    return [{
+        "name": "logs",
```
Comment (SwikritiT) [inline .drone.star:2825]: Maybe something like this?
```suggestion
        "name": "api-test-failure-logs",
```

File: tests/TestHelpers/HttpLogger.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
+<?php declare(strict_types=1);
+/**
+ * ownCloud
+ *
+ * @author Sajan Gurung <sajan@jankaritech.com>
+ * @copyright Copyright (c) 2023 Sajan Gurung sajan@jankaritech.com
```
Comment (grgprarup) [inline tests/TestHelpers/HttpLogger.php:6]:
```suggestion
 * @copyright Copyright (c) 2023, ownCloud GmbH
```
I think we can be an author, but we don't have copyright rights, that must be the owncloud.

**Review by ScharfViktor — COMMENTED** [review]
Cool, I like it  :thumbsup: 
small nitpick
 do we need add a few words to docs?

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Comment by saw-jan** [comment]
> do we need add a few words to docs?

debugging options are not documented at all in https://owncloud.dev/ocis/development/testing. So I would suggest let's do the docs part for various debugging options available in separate PR

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**